app/cell.py

```diff
@@ -1,36 +1,23 @@
-import os
 import tkinter as tk
 
 
 class Cell(tk.Button):
 
-    def __init__(self, master, row, column):
+    def __init__(self, master, row, column, background, mark):
         super().__init__(master=master)
         self.field = master
         self.app = master.parent
         self.row = str(row)
         self.column = str(column)
         self.nearby_bombs = None
         self.bomb = False
-        self.folder = os.path.dirname(__file__)
         self.state = "closed"
-        self._closed_image = tk.PhotoImage(file=(os.path.join(self.folder, "../img/closed.png")))
-        self.configure(image=self._closed_image)
-        self._last_image = tk.PhotoImage(file=(os.path.join(self.folder, "../img/boom.png")))
-        self._bomb_image = tk.PhotoImage(file=(os.path.join(self.folder, "../img/bomb.png")))
-        self._not_bomb_image = tk.PhotoImage(file=(os.path.join(self.folder, "../img/wrong.png")))
-        self._marked_image = tk.PhotoImage(file=(os.path.join(self.folder, "../img/marked.png")))
-        self._opened_image = None
-        self.bind("<Destroy>", self._on_destroy)
-
-    # Image control
-    def _get_opened_image(self):
-        return self._opened_image
-
-    def _set_opened_image(self, opened_image):
-        del self._opened_image
-        self._opened_image = opened_image
-    opened_image = property(_get_opened_image, _set_opened_image)
+        self.closed_image = background
+        self.configure(image=self.get_closed_image())
+        self.last_image = property(self.get_last_image, self.set_last_image)
+        self.bomb_image = property(self.get_bomb_image, self.set_bomb_image)
+        self.marked_image = mark
+        self.opened_image = property(self.get_opened_image, self.set_opened_image)

```

Comment:
This is not how Python properties are supposed to be used. The old version was right, with properties declared at the class level rather than in the constructor.

In the new version, you just assign a property object to a field of your class (for example, `self.last_image = property(self.get_last_image, ...)`. A consequent assignment to `self.last_image` will just override the field, without triggering `self.get_last_image` (otherwise you would get infinite recursion in your current implementation with `self.get_last_image` accessing `self.last_image`).

You can initialize these fields with `None` just as well:
```
self.last_image = None
self.bomb_image = None
self.marked_image = mark
self.opened_image = None
```